test/rules/no_legacy_typedefs.dart

```diff
@@ -0,0 +1,21 @@
+// Copyright (c) 2018, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+// test w/ `pub run test -N no_legacy_typedefs`
+
+typedef String BadNoParameters(); // LINT
+typedef bool BadOneDynamicParameter(num); // LINT
+typedef bool BadOneParameter(num value); // LINT
+typedef int BadGenericWithDynamicParameters<T>(a, b); // LINT
+typedef int BadGeneric<T>(T a, T b); // LINT
+typedef Foo BadReturnsFoo(); // LINT
```

Comment:
Maybe test one without a return type since the returntype will be `typedefKeyword.next`? Or more complex return types too?

Reply:
More tests won't hurt, of course, but given how simple I think the visit method can become, there really isn't much to test. :-)

Reply:
yea, there doesn't seem to be any more to test really